docs/config-files.md

```diff
@@ -99,10 +88,46 @@ Unfortunately, this approach can be a bit repetitive, and depending on how Babel
 could require setting ["babelrcRoots"](options.md#babelrcroots).
 
 Given that, it may be more desirable to rename the `.babelrc` to be a
-[project-wide "babel.config.js"](#project-wide-configuration). As mentioned in the project-wide
+[project-wide "babel.config.*"](#project-wide-configuration). As mentioned in the project-wide
 section above, this may then require explicitly setting ["configFile"](options.md#configfile)
 since Babel will not find the config file if the working directory isn't correct.
 
+## Supported file extensions
+
+Babel can be configured using any file extension natively supported by Node.js: you can use `.json`,
+`.js`, `.cjs` and `.mjs`, both for `babel.config.*` and `.babelrc.*` files.
+
+- `babel.config.json` and `.babelrc.json` are parsed as JSON5 and should contain an object matching
+  the [options](options.md) format that Babel accepts.
+  
+  It is recommended to choose this file type wherever possible: JS config files are
+  handy if you have complex configuration that is conditional or otherwise computed at build time.
+  However, the downside is that JS configs are less statically analyzable, and therefore have
+  negative effects on cacheability, linting, IDE autocomplete, etc.
+  Since `babel.config.json` and `.babelrc.json` are static JSON files, it allows other tools that
+  use Babel such as bundlers to cache the results of Babel safely, which can be a huge build
+  performance win.
+
+- `babel.config.cjs` and `.babelrc.cjs` allow you to define your configuration as CommonJS,
+  using `module.exports`.
+
+- `babel.config.mjs` and `.babelrc.mjs` use native ECMAScript modules. They are only supported by
```

Comment:
They are supported by Node.js 13.2+ (or older versions via the `--experimental-modules` flag)